rudder-core/src/main/scala/com/normation/rudder/hooks/RunHooks.scala

```diff
@@ -0,0 +1,194 @@
+ * The hooks are executed in the order of the list.
+ */
+
+final case class Hooks(basePath: String, hooksFile: List[String])
+/**
+ * Hook parameters are pairs of environment variable name <=> value
+ */
+final case class HookParameter(name: String, value: String)
+final case class HookParameters(values: List[HookParameter]) {
+  //shortcut to view parameters as a Map[String, String]
+  def toMap = values.map(p => (p.name, p.value)).toMap
+}
+object HookParameters {
+  def build( values: (String, String)*) = {
+    HookParameters(values.map( p => HookParameter(p._1, p._2)).toList)
+  }
+}
+
+/**
+ * Loggger for hooks
+ */
+object HooksLogger extends Logger {
+  override protected def _logger = LoggerFactory.getLogger("hooks")
+}
+
+object RunHooks {
+
+  /**
+   * Runs a list of hooks. Each hook is run sequencially (so that
+   * the user can expects one hook side effects to be used in the
+   * next one), but the whole process is asynchronous.
+   * If one hook fails, the whole list fails.
+   */
+  def asyncRun(hooks: Hooks, parameters: HookParameters): Future[Box[Unit]] = {
+    /*
+     * We can not use Future.fold, because it execute all scripts
+     * in parallele and then combine their results. Our semantic
+     * is execute script one after the other, combining at each
+     * step.
+     * But we still want the whole operation to be non-bloking.
+     */
+    ( Future(Full(()):Box[Unit]) /: hooks.hooksFile) { case (previousFuture, nextHookName) =>
+      val path = hooks.basePath + File.separator + nextHookName
+      previousFuture.flatMap {
+        case Full(())     =>
+          HooksLogger.debug(s"Run hook: '${path} ${parameters.values.mkString(" ")}'")
+          RunNuCommand.run(Cmd(path, Nil, parameters.toMap)).map { result =>
+            lazy val msg = s"Exit code ${result.code} when executing '${path} ${parameters.values.mkString(" ")}'. \n  Stdout: '${result.stdout}' \n  Stderr: '${result.stderr}'"
+            HooksLogger.trace(s"  -> results: ${msg}")
+            if(result.code != 0) {
+              Failure(msg)
+            } else {
+              Full(())
+            }
+          } recover {
+            case ex: Exception => Failure(s"Exception when executing '${path} ${parameters.values.mkString(" ")}: ${ex.getMessage}")
+          }
+        case eb: EmptyBox => Future(eb)
+      }
+    }
+  }
+
+  /*
+   * Run hooks in given directory, synchronously.
+   *
+   * Only the files with prefix ".hook" are selected as hooks, all
+   * other files will be ignored.
+   *
+   * The hooks will be run in lexigraphically order, so that the
+   * "standard" ordering of unix hooks (or init.d) with numbers
+   * works as expected:
+   *
+   * 01-first.hook
+   * 20-second.hook
+   * 30-third.hook
+   * etc
+   *
+   *
+   */
+  def syncRun(hooks: Hooks, parameters: HookParameters): Box[Unit] = {
+    try {
+      val cmdInfo = s"'${hooks.basePath}/*.hook ${parameters.values.mkString(" ")}'"
```

Comment:
It's just for comment and log, so "some_path/*.hook" is what we want to be displayed. I added a comment to explain that.